spec/predicates/vuln.md

```diff
@@ -0,0 +1,170 @@
+-   The definition of a certain set of metadata fields that could help to consume these vulnerability attestations from the different scanning tools.
+
+Obviously the first goal is quite challenging and requires a bigger community to agree upon a specific format. As a consequence, the following attestation type focuses on the definition of that common metadata which could enable the beginning of an exportable and manageable vulnerability attestation.
+
+This document describes a vulnerability attestation type to represent vulnerability reports from the scanners in an "exportable" manner and independently of the format chosen to output the results.
+
+## Prerequisites
+
+The in-toto [attestation] framework and a Vulnerability scanner tools such as [Grype](https://github.com/anchore/grype), [Trivy](https://github.com/aquasecurity/trivy) and others.
+
+## Use cases
+
+When sharing the results of a vulnerability scan using an attestation, there is certain metadata that is crucial to trust and reuse this information.
+Information about the scanner used during the scanning is relevant to trust these results. The state of the vulnerability database used to search for vulnerabilities defines the accuracy of the results. Other metadata information such as the timestamp when the scan finished could define the reusability of these results.
+
+## Model
+
+This is a predicate type that fits within the larger [Attestation] framework.
+The following model aims to provide a well defined list of fields so that consumers know how to start exchanging their scanner results.
+
+This predicate model is inspired by [cosign vulnerability attestation](https://github.com/sigstore/cosign/blob/main/specs/COSIGN_VULN_ATTESTATION_SPEC.md).
+
+## Schema
+
+The schema of this predicate type is documented below.
+
+### Fields
+
+The fields that make up this predicate type are:
+
+The `subject` contains whatever software artifacts are to be associated with this vulnerability report document.
+The `predicate` contains a JSON-encoded data with the following fields:
+
+**scanner** object
+
+> There are lots of container image scanners such as Trivy, Grype, Clair, etc.
+> This field describes which scanner is used while performing a container image scan,
+> as well as version information and which Vulnerability DB is used.
+
+**scanner.uri** string (ResourceURI), optional
+
+> > URI indicating the identity of the source of the scanner.
+
+**scanner.version** string (ResourceURI), optional
+
+> > The version of the scanner.
+
+**scanner.db.uri** string (ResourceURI), optional
+
+> > > URI indicating the identity of the source of the Vulnerability DB.
+
+**scanner.db.version** string, optional
+
+> > > The version of the Vulnerability DB.
+
+**scanner.db.lastUpdate, required** string (timestamp)
+
+> > > The timestamp of when the vulnerability DB was updated last time.
+
+**scanner.result** list
+
+> > The result contains a list of vulnerabilities.
```

Comment:
nit: can you note how readers should interpret the results if list is empty?  "An empty list indicates the scanner found no vulnerabilities."?

Related: should a note somewhere indicate what should happen if the scanner encountered an error and couldn't complete the scan?  Suggestion: don't emit the attestation at all?

Reply:
Yes, I'll add a comment to explain what an empty list means. Generally I'd prefer to emit the attestation to certify who is attesting that the scanner.version found no vulnerabilities.